### PR TITLE
`kusto_cluster` -`allowed_ips`/`fqdns`/`outbound_network_access_restricted`

### LINKED ISSUE #16420: azurerm_kusto_cluster missing parameter allowedIpRangeList
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

There's a security property on kusto cluster which is very useful, called allowedIpRangeList, it is a list of ip addresses or CIDR that are allowed access to the cluster when the cluster has internet access disabled
on az cli it looks like this:
```    --allowed-ip-range-list                                       : The list of ips in the format of
                                                                    CIDR allowed to connect to the
                                                                    cluster.
```

### New or Affected Resource(s)/Data Source(s)

azurerm_kusto_cluster

### Potential Terraform Configuration

```hcl
allowed_ip_range_list = ["123.123.123.123/24", "1.2.3.4"]
```

### References

_No response_

### PR DESCRIPTION
**Pull request by heiazuo**
### public PR for `kusto_cluster` -`allowed_ips`/`fqdns`/`outbound_network_access_restricted`

**test result after update:**

image

Fixes  #16420

### CODE REVIEW

File: internal/services/kusto/kusto_cluster_resource.go [hunk 1]
```diff
@@ -565,6 +604,9 @@ func resourceKustoClusterRead(d *pluginsdk.ResourceData, meta interface{}) error
 		d.Set("uri", props.URI)
 		d.Set("data_ingestion_uri", props.DataIngestionURI)
 		d.Set("engine", props.EngineType)
+		d.Set("allowed_ips", utils.FlattenStringSlice(props.AllowedIPRangeList))
+		d.Set("fqdns", utils.FlattenStringSlice(props.AllowedFqdnList))
+		d.Set("outbound_network_restriction_enabled", props.RestrictOutboundNetworkAccess == kusto.ClusterNetworkAccessFlagEnabled)
```
Comment (katbyte) [inline internal/services/kusto/kusto_cluster_resource.go:609]: i think in this case its best to keep this as the property name:
```suggestion
		d.Set("restrict_outbound_network_access", props.RestrictOutboundNetworkAccess == kusto.ClusterNetworkAccessFlagEnabled)
```
or
```suggestion
		d.Set("outbound_network_access_restricted", props.RestrictOutboundNetworkAccess == kusto.ClusterNetworkAccessFlagEnabled)
```

Comment (heiazuo) [reply internal/services/kusto/kusto_cluster_resource.go]: @katbyte  thanks for the great suggestions, I've updated the PR, please take another look, thanks!

**Review by stephybun — COMMENTED** [review]
There's a test failure @heiazuo 
```
------- Stdout: -------
=== RUN   TestAccKustoCluster_vnet
=== PAUSE TestAccKustoCluster_vnet
=== CONT  TestAccKustoCluster_vnet
testcase.go:110: Step 1/2 error: After applying this test step and performing a `terraform refresh`, the plan was not empty.
stdout
Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
~ update in-place
Terraform will perform the following actions:
# azurerm_network_security_rule.test_allow_management_inbound will be updated in-place
~ resource "azurerm_network_security_rule" "test_allow_management_inbound" {
id                                         = "/subscriptions/*******/resourceGroups/acctestRG-220429084634929858/providers/Microsoft.Network/networkSecurityGroups/acctestkcvozlo-nsg/securityRules/AllowAzureDataExplorerManagement"
name                                       = "AllowAzureDataExplorerManagement"
~ priority                                   = 107 -> 106
# (15 unchanged attributes hidden)
}
Plan: 0 to add, 1 to change, 0 to destroy.
--- FAIL: TestAccKustoCluster_vnet (1299.90s)
FAIL
```

File: website/docs/r/kusto_cluster.html.markdown [hunk 2]
```diff
@@ -46,14 +46,20 @@ The following arguments are supported:
 
 * `sku` - (Required) A `sku` block as defined below.
 
+* `allowed_ips` - (Optional) The list of ips in the format of CIDR allowed to connect to the cluster.
+
 * `double_encryption_enabled` - (Optional) Is the cluster's double encryption enabled? Defaults to `false`. Changing this forces a new resource to be created.
 
+* `fqdns` - (Optional) List of allowed FQDNs(Fully Qualified Domain Name) for egress from Cluster.
```
Comment (liranms) [inline website/docs/r/kusto_cluster.html.markdown:53]: how do we distinguish the cases of empty list = nothing is allowed, vs everything is allowed?

Comment (katbyte) [reply website/docs/r/kusto_cluster.html.markdown:53]: as in `fqdns = []` = nothing, and not setting `fqdns`s at all?

**Comment by liranms** [comment]
the PR looks great to me, lets merge it!

File: internal/services/kusto/kusto_cluster_resource.go [hunk 3]
```diff
@@ -137,6 +158,12 @@ func resourceKustoCluster() *pluginsdk.Resource {
 				},
 			},
 
+			"restrict_outbound_network_access": {
```
Comment (katbyte) [inline internal/services/kusto/kusto_cluster_resource.go:161]: we usually put the verb on the end pf booleans
```suggestion
			"outbound_network_access_restricted": {
```

**Review by katbyte — CHANGES_REQUESTED** [review]

**Comment by heiazuo** [comment]
Hi @katbyte，would you please take another look? Thanks!

**Review by katbyte — CHANGES_REQUESTED** [review]
three test failures to fix:

**Comment by ms-henglu** [comment]
Hi @katbyte - It seems there's only one failed case now, and I think it'll be fixed in https://github.com/hashicorp/terraform-provider-azurerm/pull/16859
image

**Comment by razsha** [comment]
@katbyte any update? we are waiting for that feature.

**Comment by tombuildsstuff** [comment]
Dependent on https://github.com/hashicorp/terraform-provider-azurerm/pull/16859?

**Pull request closed without merging**